lisa/testsuite.py

```diff
@@ -83,7 +83,12 @@ def _call_with_retry_and_timeout(
     except FunctionTimedOut:
         # FunctionTimedOut is a special exception. If it's not captured
         # explicitly, it will make the whole program exit.
-        raise TimeoutError(f"time out in {timeout} seconds.")
+        name = ""
+        # Display casename if it is timedout.
+        test_result: TestResult = test_kwargs.get("result", None)
+        if test_result:
+            name = test_result.name
+        raise TimeoutError(f"{name} time out in {timeout} seconds.")
```

Comment:
I think the log line contains name in id field already. Is it not? Can you share an example? Besides the test case name, the test case id is unique in a run.

Reply:
This is the current timeout log
`
2023-09-04 12:52:49.706[1276][ERROR] lisa.suite[Stress].case[stress_sriov_disable_enable] case failed
Traceback (most recent call last):
  File "D:\a\_work\1\s\lisa\lisa\testsuite.py", line 64, in _call_with_retry_and_timeout
    retry_call(
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\retry\api.py", line 101, in retry_call
    return __retry_internal(partial(f, *args, **kwargs), exceptions, tries, delay, max_delay, backoff, jitter, logger)
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\retry\api.py", line 33, in __retry_internal
    return f()
  File "C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\func_timeout\dafunc.py", line 101, in func_timeout
    raise FunctionTimedOut('', timeout, func, args, kwargs)
func_timeout.exceptions.FunctionTimedOut: Function stress_sriov_disable_enable (args=()) (<truncated>)}) timed out after 4500.000000 seconds.


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\a\_work\1\s\lisa\lisa\testsuite.py", line 770, in __run_case
    _call_with_retry_and_timeout(
  File "D:\a\_work\1\s\lisa\lisa\testsuite.py", line 86, in _call_with_retry_and_timeout
    raise TimeoutError(f"time out in {timeout} seconds.")
TimeoutError: time out in 4500 seconds.
`